### ISSUE TITLE
Option to restore last configured set of DNS resolvers

### ISSUE THREAD

**Issue by saradickinson** [getdnsapi/stubby#70]
When the macOS/windows scripts set the system resolvers to use local host to enable stubby they should store the current DNS configuration somewhere. 

Then when disabling stubby by reverting the system resolvers there should be an option to restore the configuration from this saved information (in addition to the default of using DHCP provided servers). 

This supports the workflow where a user wants to use a fixed set of servers (e.g. 8.8.8.8) if they can't use stubby.

**Comment by inudge** [comment]
No, I would only change the dns server. Otherwise, you'll need to do lots of work to synchronise network settings between different locations, when you're only interested in changing the one for dns. If a user changes a network setting other than dns, they would expect it to stick and not be lost when switching between strict and opportunistic dns privacy.

**Comment by inudge** [comment]
I'm working on this and have a question:
Currently, stubby-setdns-macos.sh will reset all network interfaces to use dns servers @ 127.0.0.1 ::1 regardless of whether or not IPv4 and v6 connectivity is available. Does it make any real difference ? For example, if I'm on a v4 only network and have only v4 resolver addresses configured in my stubby.yml, is there anything to be gained from telling stubby to only listen on the v4 address for dns queries and configuring the network interfaces accordingly or should I still tell it to listen on both versions of localhost regardless ?

**Comment by saradickinson** [comment]
Sorry - I missed this when you first posted! Using ::1 in this context only requires the machine to have IPv6 capabilities i.e. a link-local v6 address which gets used for routing the DNS message internally to localhost, it doesn't depend on having external IPv6 connectivity.  I believe all modern macOS version have this on by default so this shouldn't be an issue.

**Comment by inudge** [comment]
Thanks Sara. Sorry, I wasn't making myself clear enough. What I was trying to find out is whether or not there's any performance gain from only listening on the v4 interface if one knows that v6 is not available when configuring Stubby.

**Comment by saradickinson** [comment]
No - because stubby can listen locally on both IPv4 and IPv6 and then only forwards over IPv4
